src/main/java/com/jeff_media/customblockdata/BlockDataListener.java

```diff
@@ -141,6 +154,10 @@ public void onPiston(BlockPistonExtendEvent event) {
 
     @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
     public void onPiston(BlockPistonRetractEvent event) {
+        Block pistonHead = event.getBlock().getRelative(event.getDirection().getOppositeFace());
+        if(CustomBlockData.hasCustomBlockData(pistonHead, plugin)) {
+            getCbd(pistonHead).clear();
```

Comment:
Simply clearing the PDC is not a good idea. We ALWAYS want to call the CustomBlockDataRemoveEvent when doing this.

Reply:
Do you want me to do this or you got it?

Reply:
Actually, I just looked at it again, and it's a bad idea to call CustomBlockDataRemoveEvent. The only reason it is being removed is because it shouldn't be there in the first place, we lose no data by removing it. And again, this was just a quick fix I made, I didn't look into the logic of why this data is on the piston head. It might be moving the data from the block it is pushing correctly but not removing it from it's original location.